# Import Librairies

In [1]:
import numpy as np

In [4]:
divorces = pd.read_csv('divorce-csv.csv')

divorces.head()

Atr1  Atr2  Atr3  Atr4  Atr5  Atr6  Atr7  Atr8  Atr9  Atr10  ...  Atr46  \
0     2     2     4     1     0     0     0     0     0      0  ...      2   
1     4     4     4     4     4     0     0     4     4      4  ...      2   
2     2     2     2     2     1     3     2     1     1      2  ...      3   
3     3     2     3     2     3     3     3     3     3      3  ...      2   
4     2     2     1     1     1     1     0     0     0      0  ...      2   

   Atr47  Atr48  Atr49  Atr50  Atr51  Atr52  Atr53  Atr54  Class  
0      1      3      3      3      2      3      2      1      1  
1      2      3      4      4      4      4      2      2      1  
2      2      3      1      1      1      2      2      2      1  
3      2      3      3      3      3      2      2      2      1  
4      1      2      3      2      2      2      1      0      1  

[5 rows x 55 columns]

In [8]:
divorces.shape

(170, 55)

In [5]:
columns_personality = pd.read_csv('columns.csv')

columns_personality.head()

original                     short
0  I enjoy listening to music.                     Music
1                    I prefer.  Slow songs or fast songs
2           Dance, Disco, Funk                     Dance
3                   Folk music                      Folk
4                      Country                   Country

In [6]:
personalities = pd.read_csv('responses.csv')

personalities.head()


Music  Slow songs or fast songs  Dance  Folk  Country  Classical music  \
0    5.0                       3.0    2.0   1.0      2.0              2.0   
1    4.0                       4.0    2.0   1.0      1.0              1.0   
2    5.0                       5.0    2.0   2.0      3.0              4.0   
3    5.0                       3.0    2.0   1.0      1.0              1.0   
4    5.0                       3.0    4.0   3.0      2.0              4.0   

   Musical  Pop  Rock  Metal or Hardrock  ...   Age  Height  Weight  \
0      1.0  5.0   5.0                1.0  ...  20.0   163.0    48.0   
1      2.0  3.0   5.0                4.0  ...  19.0   163.0    58.0   
2      5.0  3.0   5.0                3.0  ...  20.0   176.0    67.0   
3      1.0  2.0   2.0                1.0  ...  22.0   172.0    59.0   
4      3.0  5.0   3.0                1.0  ...  20.0   170.0    59.0   

   Number of siblings  Gender  Left - right handed                Education  \
0                 1.0  female         right handed  college/bachelor degree   
1                 2.0  female         right handed  college/bachelor degree   
2                 2.0  female         right handed         secondary school   
3                 1.0  female         right handed  college/bachelor degree   
4                 1.0  female         right handed         secondary school   

   Only child  Village - town  House - block of flats  
0          no         village          block of flats  
1          no            city          block of flats  
2          no            city          block of flats  
3         yes            city          house/bungalow  
4          no         village          house/bungalow  

[5 rows x 150 columns]

In [9]:
personalities.shape


(1010, 150)

## The Data

The Kaggle dataset I used for this project is the result of a survey asked in 2013 to students.
This survey is about personalities (hobbies, musics and movies preferences, phobies, personalities traits and characteristics...).
All participants were of Slovakian nationality, aged between 15-30.

As the column name of this dataset was shortened so that I used a second dataset to have the definition of each column.

The question is how could we group this student, could we see similar paterns or correlation between data.

[Link of the dataset](https://www.kaggle.com/miroslavsabo/young-people-survey)

## Data Ingestion

In this part I will analyze deeper the dataset.

I will read the dataset and analyze

We have 1010 rows and 150 columns (139 integer and 11 categorical)

In [2]:
import pandas as pd

personalities = pd.read_csv('responses.csv')
personalities.head()

Music  Slow songs or fast songs  Dance  Folk  Country  Classical music  \
0    5.0                       3.0    2.0   1.0      2.0              2.0   
1    4.0                       4.0    2.0   1.0      1.0              1.0   
2    5.0                       5.0    2.0   2.0      3.0              4.0   
3    5.0                       3.0    2.0   1.0      1.0              1.0   
4    5.0                       3.0    4.0   3.0      2.0              4.0   

   Musical  Pop  Rock  Metal or Hardrock  ...   Age  Height  Weight  \
0      1.0  5.0   5.0                1.0  ...  20.0   163.0    48.0   
1      2.0  3.0   5.0                4.0  ...  19.0   163.0    58.0   
2      5.0  3.0   5.0                3.0  ...  20.0   176.0    67.0   
3      1.0  2.0   2.0                1.0  ...  22.0   172.0    59.0   
4      3.0  5.0   3.0                1.0  ...  20.0   170.0    59.0   

   Number of siblings  Gender  Left - right handed                Education  \
0                 1.0  female         right handed  college/bachelor degree   
1                 2.0  female         right handed  college/bachelor degree   
2                 2.0  female         right handed         secondary school   
3                 1.0  female         right handed  college/bachelor degree   
4                 1.0  female         right handed         secondary school   

   Only child  Village - town  House - block of flats  
0          no         village          block of flats  
1          no            city          block of flats  
2          no            city          block of flats  
3         yes            city          house/bungalow  
4          no         village          house/bungalow  

[5 rows x 150 columns]

In [3]:
personalities.shape

(1010, 150)

In [4]:
personalities.columns.tolist()

['Music',
 'Slow songs or fast songs',
 'Dance',
 'Folk',
 'Country',
 'Classical music',
 'Musical',
 'Pop',
 'Rock',
 'Metal or Hardrock',
 'Punk',
 'Hiphop, Rap',
 'Reggae, Ska',
 'Swing, Jazz',
 'Rock n roll',
 'Alternative',
 'Latino',
 'Techno, Trance',
 'Opera',
 'Movies',
 'Horror',
 'Thriller',
 'Comedy',
 'Romantic',
 'Sci-fi',
 'War',
 'Fantasy/Fairy tales',
 'Animated',
 'Documentary',
 'Western',
 'Action',
 'History',
 'Psychology',
 'Politics',
 'Mathematics',
 'Physics',
 'Internet',
 'PC',
 'Economy Management',
 'Biology',
 'Chemistry',
 'Reading',
 'Geography',
 'Foreign languages',
 'Medicine',
 'Law',
 'Cars',
 'Art exhibitions',
 'Religion',
 'Countryside, outdoors',
 'Dancing',
 'Musical instruments',
 'Writing',
 'Passive sport',
 'Active sport',
 'Gardening',
 'Celebrities',
 'Shopping',
 'Science and technology',
 'Theatre',
 'Fun with friends',
 'Adrenaline sports',
 'Pets',
 'Flying',
 'Storm',
 'Darkness',
 'Heights',
 'Spiders',
 'Snakes',
 'Rats',
 'Agein

In [5]:
personalities.dtypes

Music                         float64
Slow songs or fast songs      float64
Dance                         float64
Folk                          float64
Country                       float64
Classical music               float64
Musical                       float64
Pop                           float64
Rock                          float64
Metal or Hardrock             float64
Punk                          float64
Hiphop, Rap                   float64
Reggae, Ska                   float64
Swing, Jazz                   float64
Rock n roll                   float64
Alternative                   float64
Latino                        float64
Techno, Trance                float64
Opera                         float64
Movies                        float64
Horror                        float64
Thriller                      float64
Comedy                        float64
Romantic                      float64
Sci-fi                        float64
War                           float64
Fantasy/Fair

In [6]:
null_cols = personalities.isnull().sum()

print(null_cols[null_cols > 0])

Music                          3
Slow songs or fast songs       2
Dance                          4
Folk                           5
Country                        5
Classical music                7
Musical                        2
Pop                            3
Rock                           6
Metal or Hardrock              3
Punk                           8
Hiphop, Rap                    4
Reggae, Ska                    7
Swing, Jazz                    6
Rock n roll                    7
Alternative                    7
Latino                         8
Techno, Trance                 7
Opera                          1
Movies                         6
Horror                         2
Thriller                       1
Comedy                         3
Romantic                       3
Sci-fi                         2
War                            2
Fantasy/Fairy tales            3
Animated                       3
Documentary                    8
Western                        4
          

In [9]:
null_rows = personalities[personalities.isnull().any(axis=1)]
pd.options.display.max_columns = 150

null_rows


Music  Slow songs or fast songs  Dance  Folk  Country  Classical music  \
3       5.0                       3.0    2.0   1.0      1.0              1.0   
8       5.0                       3.0    3.0   1.0      1.0              2.0   
15      1.0                       3.0    2.0   2.0      3.0              4.0   
17      5.0                       3.0    3.0   3.0      3.0              2.0   
22      5.0                       3.0    3.0   2.0      3.0              3.0   
27      4.0                       5.0    3.0   4.0      1.0              3.0   
37      5.0                       5.0    1.0   1.0      1.0              5.0   
45      5.0                       4.0    5.0   2.0      2.0              4.0   
46      5.0                       3.0    4.0   3.0      4.0              1.0   
47      5.0                       3.0    3.0   1.0      1.0              2.0   
51      5.0                       3.0    3.0   3.0      1.0              4.0   
56      5.0                       3.0    1.0   2.0      3.0              4.0   
58      5.0                       5.0    5.0   3.0      2.0              3.0   
63      5.0                       4.0    NaN   1.0      3.0              2.0   
69      5.0                       4.0    1.0   1.0      1.0              1.0   
72      5.0                       3.0    2.0   2.0      2.0              2.0   
76      4.0                       4.0    4.0   1.0      1.0              2.0   
78      4.0                       4.0    4.0   2.0      4.0              1.0   
83      5.0                       3.0    5.0   1.0      2.0              4.0   
84      5.0                       3.0    1.0   1.0      4.0              5.0   
87      5.0                       4.0    4.0   1.0      2.0              2.0   
91      5.0                       5.0    1.0   5.0      1.0              NaN   
93      5.0                       4.0    1.0   5.0      5.0              4.0   
94      5.0                       5.0    5.0   1.0      1.0              1.0   
102     5.0                       3.0    1.0   2.0      3.0              5.0   
103     5.0                       5.0    5.0   2.0      2.0              NaN   
107     5.0                       3.0    1.0   1.0      1.0              2.0   
113     3.0                       4.0    4.0   2.0      3.0              1.0   
115     5.0                       4.0    5.0   2.0      2.0              5.0   
124     4.0                       4.0    1.0   4.0      NaN              5.0   
130     4.0                       2.0    2.0   1.0      1.0              4.0   
137     5.0                       3.0    4.0   3.0      3.0              5.0   
138     5.0                       3.0    2.0   3.0      3.0              4.0   
140     5.0                       4.0    3.0   2.0      1.0              2.0   
142     5.0                       4.0    4.0   1.0      3.0              2.0   
143     5.0                       4.0    4.0   2.0      1.0              1.0   
144     5.0                       2.0    3.0   1.0      2.0              3.0   
145     1.0                       4.0    4.0   1.0      1.0              2.0   
146     5.0                       3.0    5.0   2.0      1.0              4.0   
149     2.0                       3.0    1.0   3.0      1.0              3.0   
153     2.0                       5.0    5.0   2.0      2.0              3.0   
159     5.0                       3.0    4.0   4.0      3.0              5.0   
162     4.0                       2.0    3.0   5.0      4.0              5.0   
164     5.0                       4.0    4.0   2.0      1.0              3.0   
167     3.0                       3.0    3.0   2.0      2.0              NaN   
170     4.0                       3.0    2.0   3.0      2.0              2.0   
174     5.0                       2.0    1.0   5.0      3.0              3.0   
176     5.0                       3.0    3.0   NaN      1.0              1.0   
177     5.0                       3.0    1.0   4.0      2.0              4.0   
180   

For MUSIC PREFERENCES, MOVIE PREFERENCES, HOBBIES & INTERESTS, PHOBIAS, SPENDING HABITS we will replace NaN Value by 0 as "no answer".

We will first replace all the NaN value by 0 and then proceed an other way for special columns as Age, Village, only child ...

Create a function to replace by 0 if 1 to 5

By 0 or 1 if yes/no



## Data Wrangling and Cleaning